Index: ChangeLog
===================================================================
--- ChangeLog	(revision 240406)
+++ ChangeLog	(working copy)
@@ -1,5 +1,12 @@
 2009-03-17  Don McCoy  <don@codesourcery.com>
 
+	* src/vsip/initfin.cpp (finalize_default_pool): Use new function 
+	  to free default pool when finalizing library.
+	* src/vsip/core/memory_pool.hpp: Declare it.
+	* src/vsip/core/memory_pool.cpp: Define it.
+
+2009-03-17  Don McCoy  <don@codesourcery.com>
+
 	* src/vsip/core/type_list.hpp: Lengthened the type list.
 	* src/vsip/opt/diag/eval.hpp: Added CUDA tag.
 	* src/vsip/opt/cuda/bindings.hpp: Added wrapper functions
Index: src/vsip/initfin.cpp
===================================================================
--- src/vsip/initfin.cpp	(revision 240405)
+++ src/vsip/initfin.cpp	(working copy)
@@ -262,5 +262,7 @@
 
   delete profiler_opts_;
   profiler_opts_ = 0;
+
+  impl::finalize_default_pool();
 #endif
 }
Index: src/vsip/core/memory_pool.hpp
===================================================================
--- src/vsip/core/memory_pool.hpp	(revision 240405)
+++ src/vsip/core/memory_pool.hpp	(working copy)
@@ -63,6 +63,7 @@
 
 void initialize_default_pool(int& argc, char**&argv);
 
+void finalize_default_pool();
 
 
 } // namespace vsip::impl
Index: src/vsip/core/memory_pool.cpp
===================================================================
--- src/vsip/core/memory_pool.cpp	(revision 240405)
+++ src/vsip/core/memory_pool.cpp	(working copy)
@@ -38,6 +38,12 @@
   default_pool = new Aligned_pool();
 }
 
+void finalize_default_pool()
+{
+  delete default_pool;
+}
+
+
 } // namespace vsip::impl
 
 } // namespace vsip
