Index: qm/attachment.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/attachment.py,v
retrieving revision 1.17
diff -u -r1.17 attachment.py
--- qm/attachment.py	23 Jun 2003 06:46:47 -0000	1.17
+++ qm/attachment.py	3 Jul 2003 12:45:13 -0000
@@ -191,12 +191,6 @@
 
         pass
 
-
-    def GetIndex(self):
-        """Returns the index associated with this attachment store."""
-
-        return self.__index
-
     
     def GetData(self, location):
         """Return the data for an attachment.
Index: qm/fields.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/fields.py,v
retrieving revision 1.72
diff -u -r1.72 fields.py
--- qm/fields.py	23 Jun 2003 23:42:09 -0000	1.72
+++ qm/fields.py	3 Jul 2003 12:45:14 -0000
@@ -1243,7 +1243,6 @@
                     value.GetDescription(),
                     value.GetMimeType(),
                     value.GetLocation(),
-                    ("%d" % value.GetStore().GetIndex()),
                     value.GetFileName(),
                     )
                 # Each part is URL-encoded.
Index: qm/test/classes/xml_database.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/test/classes/xml_database.py,v
retrieving revision 1.12
diff -u -r1.12 xml_database.py
--- qm/test/classes/xml_database.py	23 Jun 2003 06:46:47 -0000	1.12
+++ qm/test/classes/xml_database.py	3 Jul 2003 12:45:14 -0000
@@ -187,9 +187,9 @@
         # Is the file name by itself OK in this directory?  It must not
         # have a file extension used by the XML database itself, and
         # there must be no other file with the same name there.
-        if extension not in [self.__database.GetTestExtension(),
-                             self.__database.GetSuiteExtension(),
-                             self.__database.GetResourceExtension()] \
+        if extension not in [self.GetTestExtension(),
+                             self.GetSuiteExtension(),
+                             self.GetResourceExtension()] \
            and not os.path.exists(full_data_file_path):
             return data_file_path
 
