Index: qm/test/web/web.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/test/web/web.py,v
retrieving revision 1.70
diff -u -r1.70 web.py
--- qm/test/web/web.py	3 Jul 2003 19:32:04 -0000	1.70
+++ qm/test/web/web.py	14 Jul 2003 07:19:38 -0000
@@ -1212,6 +1212,7 @@
         self.__resource_results = {}
         # The stream is not finished yet.
         self.__is_finished = 0
+        self.__annotations = {}
         
         # Create a lock for synchronization between the test execution
         # thread (which will call methods like 'WriteResults' and
