Index: qm/test/web/web.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/test/web/web.py,v
retrieving revision 1.70
diff -u -r1.70 web.py
--- qm/test/web/web.py	3 Jul 2003 19:32:04 -0000	1.70
+++ qm/test/web/web.py	14 Jul 2003 07:33:40 -0000
@@ -433,7 +433,8 @@
         if directory == "":
             return self.test_results.values()
         
-        return filter(lambda r: self.__IsLabelInDirectory(r.GetId()),
+        return filter(lambda r: self.__IsLabelInDirectory(r.GetId(),
+                                                          directory),
                       self.test_results.values())
                       
 
