Index: qm/test/web/web.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/test/web/web.py,v
retrieving revision 1.70
diff -u -r1.70 web.py
--- qm/test/web/web.py	3 Jul 2003 19:32:04 -0000	1.70
+++ qm/test/web/web.py	14 Jul 2003 07:54:55 -0000
@@ -706,12 +706,12 @@
 
         returns -- True if 'id' indicates a test contained in
         'directory', or one of its subdirectories."""
-        
-        while len(id) >= directory:
+
+        while len(id) >= len(directory):
             if id == directory:
                 return 1
             id = self.GetDatabase().SplitLabel(id)[0]
-
+            
         return 0
 
 
