Index: ChangeLog
===================================================================
--- ChangeLog	(revision 228901)
+++ ChangeLog	(working copy)
@@ -1,3 +1,8 @@
+2008-11-23  Don McCoy  <don@codesourcery.com>
+
+	* tests/tutorial/*.cpp: Corrected directory path doc/tutorial to
+	  doc/users-guide.
+
 2008-11-20  Jules Bergmann  <jules@codesourcery.com>
 
 	Fix ukernel/interp.cpp 64-bit failure:
Index: tests/tutorial/matlab_iter_example.cpp
===================================================================
--- tests/tutorial/matlab_iter_example.cpp	(revision 228901)
+++ tests/tutorial/matlab_iter_example.cpp	(working copy)
@@ -16,7 +16,7 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/matlab_iter_example_prelude.cpp>
+#include <../doc/users-guide/src/matlab_iter_example_prelude.cpp>
 
 #include <vsip_csl/test.hpp>
 #include <vsip_csl/error_db.hpp>
@@ -25,13 +25,13 @@
 void
 test_write()
 {
-#include <../doc/tutorial/src/matlab_bin_formatter_write.cpp>
+#include <../doc/users-guide/src/matlab_bin_formatter_write.cpp>
 }
 
 void
 test_read_matrix()
 {
-#include <../doc/tutorial/src/matlab_iter_example1.cpp>
+#include <../doc/users-guide/src/matlab_iter_example1.cpp>
 
   // Check result.  Example reads record 'm' from file into matrix 'm'.
   Matrix<float> chk_m(3, 3);
@@ -46,7 +46,7 @@
 void
 test_read_vector()
 {
-#include <../doc/tutorial/src/matlab_iter_example2.cpp>
+#include <../doc/users-guide/src/matlab_iter_example2.cpp>
 
   // Check result.  Example reads record 'v' from file into vector 'v'.
   Vector<float> chk_v(3);
Index: tests/tutorial/mbf_example.cpp
===================================================================
--- tests/tutorial/mbf_example.cpp	(revision 228901)
+++ tests/tutorial/mbf_example.cpp	(working copy)
@@ -16,7 +16,7 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/matlab_bin_formatter_prelude.cpp>
+#include <../doc/users-guide/src/matlab_bin_formatter_prelude.cpp>
 
 #include <vsip_csl/test.hpp>
 #include <vsip_csl/error_db.hpp>
@@ -25,13 +25,13 @@
 void
 test_write()
 {
-#include <../doc/tutorial/src/matlab_bin_formatter_write.cpp>
+#include <../doc/users-guide/src/matlab_bin_formatter_write.cpp>
 }
 
 void
 test_read()
 {
-#include <../doc/tutorial/src/matlab_bin_formatter_read.cpp>
+#include <../doc/users-guide/src/matlab_bin_formatter_read.cpp>
 
   // Initialize matrix 'm'.
   Matrix<float> chk_m(3, 3);
Index: tests/tutorial/mtf_example.cpp
===================================================================
--- tests/tutorial/mtf_example.cpp	(revision 228901)
+++ tests/tutorial/mtf_example.cpp	(working copy)
@@ -16,13 +16,13 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/matlab_text_formatter_prelude.cpp>
+#include <../doc/users-guide/src/matlab_text_formatter_prelude.cpp>
 
 
 void
 test()
 {
-#include <../doc/tutorial/src/matlab_text_formatter.cpp>
+#include <../doc/users-guide/src/matlab_text_formatter.cpp>
 }
 
 
Index: tests/tutorial/par_fc2_foreach.cpp
===================================================================
--- tests/tutorial/par_fc2_foreach.cpp	(revision 228901)
+++ tests/tutorial/par_fc2_foreach.cpp	(working copy)
@@ -16,4 +16,4 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/par/fc2-foreach.cpp>
+#include <../doc/users-guide/src/par/fc2-foreach.cpp>
Index: tests/tutorial/par_fc2_local_views.cpp
===================================================================
--- tests/tutorial/par_fc2_local_views.cpp	(revision 228901)
+++ tests/tutorial/par_fc2_local_views.cpp	(working copy)
@@ -15,4 +15,4 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/par/fc2-local-views.cpp>
+#include <../doc/users-guide/src/par/fc2-local-views.cpp>
Index: tests/tutorial/par_fc3_io.cpp
===================================================================
--- tests/tutorial/par_fc3_io.cpp	(revision 228901)
+++ tests/tutorial/par_fc3_io.cpp	(working copy)
@@ -15,4 +15,4 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/par/fc3-io.cpp>
+#include <../doc/users-guide/src/par/fc3-io.cpp>
Index: tests/tutorial/par_fc1_parallel.cpp
===================================================================
--- tests/tutorial/par_fc1_parallel.cpp	(revision 228901)
+++ tests/tutorial/par_fc1_parallel.cpp	(working copy)
@@ -15,4 +15,4 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/par/fc1-parallel.cpp>
+#include <../doc/users-guide/src/par/fc1-parallel.cpp>
Index: tests/tutorial/par_fc1_serial.cpp
===================================================================
--- tests/tutorial/par_fc1_serial.cpp	(revision 228901)
+++ tests/tutorial/par_fc1_serial.cpp	(working copy)
@@ -15,4 +15,4 @@
   Included Files
 ***********************************************************************/
 
-#include <../doc/tutorial/src/par/fc1-serial.cpp>
+#include <../doc/users-guide/src/par/fc1-serial.cpp>
Index: tests/tutorial/profile_example.cpp
===================================================================
--- tests/tutorial/profile_example.cpp	(revision 228901)
+++ tests/tutorial/profile_example.cpp	(working copy)
@@ -71,5 +71,5 @@
   // frequency domain
   for_fft(replica);
 
-  #include <../doc/tutorial/src/profile_example.cpp>
+  #include <../doc/users-guide/src/profile_example.cpp>
 }
Index: tests/tutorial/par_read_file.cpp
===================================================================
--- tests/tutorial/par_read_file.cpp	(revision 228901)
+++ tests/tutorial/par_read_file.cpp	(working copy)
@@ -33,8 +33,8 @@
   Main Program
 ***********************************************************************/
 
-#include <../doc/tutorial/src/par/read_file.hpp>
-#include <../doc/tutorial/src/par/write_file.hpp>
+#include <../doc/users-guide/src/par/read_file.hpp>
+#include <../doc/users-guide/src/par/write_file.hpp>
 
 int
 main(int argc, char** argv)
