Index: ChangeLog
===================================================================
--- ChangeLog	(revision 228902)
+++ ChangeLog	(working copy)
@@ -1,5 +1,9 @@
 2008-11-23  Don McCoy  <don@codesourcery.com>
 
+	* src/vsip/core/matvec_prod.hpp: Deleted unused typedefs.
+
+2008-11-23  Don McCoy  <don@codesourcery.com>
+
 	* tests/tutorial/*.cpp: Corrected directory path doc/tutorial to
 	  doc/users-guide.
 
Index: src/vsip/core/matvec_prod.hpp
===================================================================
--- src/vsip/core/matvec_prod.hpp	(revision 228901)
+++ src/vsip/core/matvec_prod.hpp	(working copy)
@@ -212,8 +212,6 @@
   const_Matrix<T1, Block1> b,
   Matrix<T2, Block2>       r)
 {
-  typedef Op_list_2<Block0, Block1> signature_type;
-
   assert(r.size(0) == a.size(0));
   assert(r.size(1) == b.size(1));
   assert(a.size(1) == b.size(0));
@@ -243,8 +241,6 @@
   const_Matrix<T1, Block1> b,
   Matrix<T2, Block2>       r)
 {
-  typedef Op_list_2<Block0, Block1> signature_type;
-
   assert(r.size(0) == a.size(0));
   assert(r.size(1) == b.size(1));
   assert(a.size(1) == b.size(0));
@@ -272,8 +268,6 @@
   const_Vector<T1, Block1> b,
   Vector<T2, Block2>       r)
 {
-  typedef Op_list_2<Block0, Block1> signature_type;
-
   assert(r.size() == a.size(0));
   assert(a.size(1) == b.size());
 
@@ -302,8 +296,6 @@
   const_Matrix<T1, Block1> b,
   Vector<T2, Block2>       r)
 {
-  typedef Op_list_2<Block0, Block1> signature_type;
-
   assert(r.size() == b.size(1));
   assert(a.size() == b.size(0));
 
