Index: qm/test/web/web.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/test/web/web.py,v
retrieving revision 1.70
diff -u -r1.70 web.py
--- qm/test/web/web.py	3 Jul 2003 19:32:04 -0000	1.70
+++ qm/test/web/web.py	15 Jul 2003 05:29:22 -0000
@@ -1887,7 +1887,7 @@
         rs = rsc({ "file" : s })
         # Write all the results.
         for (id, outcome) in self.__expected_outcomes.items():
-            r = Result(Result.TEST, id, Context(), outcome)
+            r = Result(Result.TEST, id, outcome)
             rs.WriteResult(r)
         # Terminate the stream.
         rs.Summarize()
