Index: qm/attachment.py
===================================================================
RCS file: /home/qm/Repository/qm/qm/attachment.py,v
retrieving revision 1.19
diff -u -r1.19 attachment.py
--- qm/attachment.py	15 Jul 2003 18:48:21 -0000	1.19
+++ qm/attachment.py	17 Jul 2003 11:55:32 -0000
@@ -277,8 +277,8 @@
 
     def GetDataFile(self, location):
 
-        if root is not None:
-            return os.path.join(root, location)
+        if self.__root is not None:
+            return os.path.join(self.__root, location)
         else:
             return location
 
